typing/ctype.ml

```diff
@@ -2690,13 +2681,35 @@ let rec unify (env:Env.t ref) t1 t2 =
     reset_trace_gadt_instances reset_tracing;
     raise( Unify (Errortrace.diff t1 t2 :: trace) )
 
-and unify2 env t1 t2 =
+and unify2 env t1 t2 = unify2_expand env t1 t1 t2 t2
+
+and unify2_rec env t10 t1 t20 t2 =
+  let t1 = repr t1 and t2 = repr t2 in
+  if unify_eq t1 t2 then () else
+  try match (t1.desc, t2.desc) with
+  | (Tconstr (p1, tl1, a1), Tconstr (p2, tl2, a2)) ->
+      if Path.same p1 p2 && tl1 = [] && tl2 = [] && is_public_type !env p1
```

Comment:
I don't see why we still need the `is_public_type` test here.

Reply:
Actually, this is mostly conservatism.
`is_public_type` only checks `type_kind` and `type_private`, and if it fails for either reason the type cannot be expanded anyway, so we will just end up doing the same unification in `unify3`.
Maybe it is ok to just remove it.

Reply:
I think it would be best to just remove it. It makes the code harder to think about and doesn't really add anything.